### PR TITLE
Add RestrictTo annotations to Firestore

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Same as #266 and #270 but for Firestore.

I fixed a couple of places where we were missing the `@PublicApi` annotation. See cr/151329304 (internal) for reference.

Addresses #264

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java [hunk 1]
```diff
@@ -71,6 +72,7 @@
   }
 
   /** @hide */
+  @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
```
Comment (rsgowman) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java]: We recently dropped the 'public' from this, so we don't need to explicitly "RestrictTo" it. i.e. if you merge master, you can revert your changes to this file.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentSnapshot.java [hunk 2]
```diff
@@ -313,6 +314,7 @@ public Object get(
    * @return The value at the given field or null.
    */
   @Nullable
+  @PublicApi
```
Comment (rsgowman) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentSnapshot.java]: Note that despite lacking this annotation, these methods were still showing up in our public api.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/SetOptions.java [hunk 3]
```diff
@@ -46,12 +47,14 @@ private SetOptions(boolean merge, @Nullable FieldMask fieldMask) {
   }
 
   /** @hide */
+  @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
```
Comment (rsgowman) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/SetOptions.java]: Similar to above, this method is no longer public (but note that getFIeldMask() still remains as a public method.)

File: firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java (hunk 1 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java]: Merged and removed.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentSnapshot.java (hunk 2 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentSnapshot.java]: `@PublicApi` was used when we obfuscated our codebase. I am wondering if Proguard does not obfuscate methods when overloads are marked `@PublicApi`. But we hopefully never have to find an answer to that question :)

File: firebase-firestore/src/main/java/com/google/firebase/firestore/SetOptions.java (hunk 3 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/SetOptions.java]: Done

**Pull request merged**